src/ValueFormatters/QuantityFormatter.php

```diff
@@ -158,10 +158,17 @@ protected function formatNumber( QuantityValue $quantity ) {
 		$roundingExponent = $this->getRoundingExponent( $quantity );
 
 		$amount = $quantity->getAmount();
-		$roundedAmount = $this->decimalMath->roundToExponent( $amount, $roundingExponent );
-		$formatted = $this->decimalFormatter->format( $roundedAmount );
 
-		$margin = $this->formatMargin( $quantity->getUncertaintyMargin(), $roundingExponent );
+		if ( $roundingExponent === null ) {
```

Comment:
Keep in mind that UnboundedQuantityValues should only be rounded if that is explicitly requested by the user.